### PR TITLE
#570 Add OSET Public License & its test

### PR DESCRIPTION
**Pull request by yasharmaster**
Signed-off-by: Yash Sharma <yasharmaster@gmail.com>

### CODE REVIEW

**Comment by yasharmaster** [comment]
@pombredanne please review

File: src/licensedcode/data/licenses/osetpl-2.1.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+key: osetpl-2.1
+short_name: osetpl 2.1
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/osetpl-2.1.yml]: Use OSET 2.1 instead

Comment (pombredanne) [inline src/licensedcode/data/licenses/osetpl-2.1.yml]: just for the short name

**Comment by yasharmaster** [comment]
@pombredanne I have updated the short name.

File: src/licensedcode/data/licenses/osetpl-2.1.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+key: osetpl-2.1
+short_name: OSET 2.1
+name: OSET Public License v2.1
+
+category: Proprietary Free
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/osetpl-2.1.yml]: I  wonder if this is a proprietary license or a plain FOSS. This is after all derived from the MPL 2.0. I guess I could reach out to the author...

Comment (heathermeeker) [inline src/licensedcode/data/licenses/osetpl-2.1.yml]: OSET PL is similar to Mozilla, with minor alterations to support government contracting.  It is OSI approved.

Comment (pombredanne) [inline src/licensedcode/data/licenses/osetpl-2.1.yml]: @heathermeeker Thanks you ++ for your input!

Comment (pombredanne) [inline src/licensedcode/data/licenses/osetpl-2.1.yml]: @yasharmaster Heather authored this license. So please use the same category as for the MPL 2.0

**Comment by yasharmaster** [comment]
@pombredanne I have changed the category to 'Copyleft Limited' which is also the category of MPL 2.0.

**Comment by pombredanne** [comment]
LGTM. Thanks and thanks to @heathermeeker for chiming in!

**Pull request merged**